src/menu.rs

```diff
@@ -6,14 +6,57 @@ use std::{
   hash::{Hash, Hasher},
 };
 
+use crate::platform_impl::{MenuBuilder as PlatformMenuBuilder, MenuItem};
+
+#[derive(Debug, Clone)]
+pub struct MenuBuilder(pub(crate) PlatformMenuBuilder);
+
+impl MenuBuilder {
+  pub fn init() -> Self {
```

Comment:
`MenuType` should be passed here instead imho.
Also, `new` is a more common constructor name in rust.